src/DynamoCoreWpf/UI/Converters.cs

```diff
@@ -1840,18 +1840,59 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
         }
     }
 
-    public class NumberFormatToBoolConverter : IValueConverter
+    public class BinaryRadioButtonCheckedConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter,
+            System.Globalization.CultureInfo culture)
+        {
+            return value.Equals(bool.Parse(parameter.ToString()));
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter,
+            System.Globalization.CultureInfo culture)
+        {
+            return value.Equals(true) ? bool.Parse(parameter.ToString()) : Binding.DoNothing;
+        }
+    }
+
+    public class NumberFormatConverter : IValueConverter
     {
         public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
         {
-            if (parameter.ToString() == SIUnit.NumberFormat)
-                return true;
-            return false;
+            switch (value)
+            {
+                case "f0":
+                    return Resources.DynamoViewSettingMenuNumber0;
+                case "f1":
+                    return Resources.DynamoViewSettingMenuNumber00;
+                case "f2":
+                    return Resources.DynamoViewSettingMenuNumber000;
+                case "f3":
+                    return Resources.DynamoViewSettingMenuNumber0000;
+                case "f4":
+                    return Resources.DynamoViewSettingMenuNumber00000;
+                default:
+                    return null;
```

Comment:
I feel we should also have a bunch of legacy code we can obsolete/ delete here? The old number format comes with a dialog, let's delete it?

Reply:
Hello Aaron, can you explain further? Previously the node format was chosen from a sub-menu in the Settings Dynamo menu which we removed already. This preference is saved on the PreferencesSettings with the codes "f1", "f2", etc but shown to the user as 0.0, 0.00, etc, so this converter is used to convert back and forth between the two formats.

Reply:
ah OK, my bad then that I did not notice we are already re-using the resources